l10n_fr_fec_oca/wizard/account_fr_fec_oca.py

```diff
@@ -0,0 +1,579 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+
+import base64
+import io
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools import float_is_zero
+import logging
+logger = logging.getLogger(__name__)
+
+try:
+    from unidecode import unidecode
+except ImportError:
+    logger.debug('Cannot import unidecode')
+try:
+    import unicodecsv
+except ImportError:
+    logger.debug('Cannot import unicodecsv')
+
+
+class AccountFrFecOca(models.TransientModel):
+    _name = 'account.fr.fec.oca'
+    _description = 'Ficher Echange Informatise'
+
+    date_range_id = fields.Many2one('date.range', string='Date Range')
+    date_from = fields.Date(string='Start Date', required=True)
+    date_to = fields.Date(string='End Date', required=True)
+    encoding = fields.Selection([
+        ('iso8859_15', 'ISO-8859-15'),
+        ('utf-8', 'UTF-8'),
+        # ('cp500', 'EBCDIC'),
+        ('ascii', 'ASCII'),
+        ], string='Encoding', default='iso8859_15', required=True)
+    delimiter = fields.Selection([
+        ('|', '|'),
+        ('tab', 'Tab'),
+        ], default='|', string='Field Delimiter', required=True)
+    partner_option = fields.Selection([
+        ('types', 'Selected Account Types'),
+        ('accounts', 'Selected Accounts'),
+        ('all', 'All'),
+        ], default='types', required=True, string='Partner Export Option')
+    partner_account_type_ids = fields.Many2many(
+        'account.account.type', string='Account Types',
+        default=lambda self: self._default_partner_account_type_ids())
+    partner_account_ids = fields.Many2many(
+        'account.account', string='Accounts',
+        default=lambda self: self._default_partner_account_ids())
+    fec_data = fields.Binary('FEC File', readonly=True, attachment=False)
+    filename = fields.Char(string='Filename', size=256, readonly=True)
+    export_type = fields.Selection([
+        ('official', 'Official FEC report (posted entries only)'),
+        ('nonofficial',
+         'Non-official FEC report (posted and unposted entries)'),
+        ], string='Export Type', required=True, default='official')
+
+    @api.onchange('date_range_id')
+    def date_range_change(self):
+        if self.date_range_id:
+            self.date_from = self.date_range_id.date_start
+            self.date_to = self.date_range_id.date_end
+
+    @api.model
+    def _default_partner_account_type_ids(self):
+        receivable = self.env.ref('account.data_account_type_receivable')
+        payable = self.env.ref('account.data_account_type_payable')
+        return receivable + payable
+
+    @api.model
+    def _default_partner_account_ids(self):
+        pay = self.env['ir.property']._get(
+            'property_account_payable_id', 'res.partner')
+        rec = self.env['ir.property']._get(
+            'property_account_receivable_id', 'res.partner')
+        return pay + rec
+
+    def do_query_unaffected_earnings(self):
```

Comment:
il serait bien de mettre cette méthode en privée (du fait que c'est du pur SQL) + rajouter un `hasgroup`.

Reply:
Now that we have ir.model.access on wizards in v14, I think it's not usefull to have a hasgroup(), no ? I agree that it would be useful on v13-.